File: complaints_management/complaints_management/report/complaint_statistics/complaint_statistics.js
// Copyright (c) 2022, Josef Engelskirchen and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Complaint Statistics"] = {
	"filters": [
           {
              "fieldname":"posting_date",
              "label":("From Date"),
              "fieldtype":"Date",
              "default":get_today(),
             "reqd":1
          },
           {
              "fieldname":"posting_date",
              "label":("To Date"),
              "fieldtype":"Date",
              "default":get_today(),
             "reqd":1
          },
           {
              "fieldname":"supplier",
              "label":("Supplier"),
              "fieldtype":"Link",
              "options":"Supplier"
          },
           {
              "fieldname":"form_types",
              "label":("Form Type"),
              "fieldtype":"Select",
              "options":["","C-Form",H-Form"]
          }
	]
};

File: complaints_management/complaints_management/report/complaint_statistics/complaint_statistics.py
# Copyright (c) 2022, Josef Engelskirchen and contributors
# For license information, please see license.txt

import frappe
import datetime

def execute(filters=None):
    columns = [
        {'fieldname':'date','label':'Complaint Date','fieldtype':'Date'},
        {'fieldname':'name','label':'Complaint Nr','fieldtype':'Data'},
	{'fieldname':'customer','label':'Customer','fieldtype':'Data'}
    ]
    data = frappe.db.get_all('Complaint', ['date','name','customer'])
#    message = ["The equivalent of the letters 'cats' in numbers is '2287'"]
    return columns, data
